Cron/Order.php

```diff
@@ -277,4 +287,28 @@ protected function addOrderComment(OrderInterface $order, string $status, string
 
         return $returnStatus;
     }
+
+    /**
+     * Get an Order from an order increment id. Attempts several retries due to Magento lag issues.
+     *
+     * @param string $orderIncrementId
+     *
+     * @return OrderInterface|null An order
+     */
+    protected function getOrderByIncrementId(string $incrementId) : ?OrderInterface
+    {
+        $orderFetchCount = 0;
+        do {
+            $order =  $this->orderHelper->getOrderByIncrementId($incrementId);
+            if ($order && $order->getId()) {
+                return $order;
+            }
+
+            $orderFetchCount++;
+            // phpcs:ignore
+            sleep(self::ORDER_FETCH_SLEEP_TIME);
+        } while ($orderFetchCount <= self::MAX_ORDER_FETCH_ATTEMPTS);
+
+        return $order;
```

Comment:
Maybe should just explicitly return `null` here? We know that `$order` isn't really an order if we made it out of the loop.